### ISSUE TITLE
[DEFECT] Error starting on non-Windows platforms

### ISSUE THREAD

**Issue by dhill2522** [idaholab/raven#1099]
--------
Defect Description
--------

**Error Starting RAVEN on Linux**

##### What did you expect to see happen?
On running `./raven_framework interactive` I expected RAVEN to start in interactive mode.

##### What did you see instead?
RAVEN failed to start and the following traceback was given.

```
Traceback (most recent call last):
  File "/home/dhill25/repos/raven/framework/Driver.py", line 52, in <module>
    from Simulation import Simulation
  File "/home/dhill25/repos/raven/framework/Simulation.py", line 36, in <module>
    import Steps
  File "/home/dhill25/repos/raven/framework/Steps.py", line 46, in <module>
    import Models
  File "/home/dhill25/repos/raven/framework/Models/__init__.py", line 32, in <module>
    from .PostProcessor import PostProcessor
  File "/home/dhill25/repos/raven/framework/Models/PostProcessor.py", line 31, in <module>
    import PostProcessors
  File "/home/dhill25/repos/raven/framework/PostProcessors/__init__.py", line 31, in <module>
    from .DataMining import DataMining
  File "/home/dhill25/repos/raven/framework/PostProcessors/DataMining.py", line 34, in <module>
    import unSupervisedLearning
  File "/home/dhill25/repos/raven/framework/unSupervisedLearning.py", line 59, in <module>
    matplotlib.use('TkAgg')
NameError: name 'matplotlib' is not defined
```

##### Do you have a suggested fix for the development team?
It appears that a recent commit (72d763a234c2483e10f5aaebc55a77a879de2cf2) broke the framework on non-Windows platforms. It appears to be a simple lack of importing matplotlib before setting the backend.

It can be fixed very easily by adding `import matplotlib` sometime before line 59 in `framework/unSupervisedLearning.py`.

**Describe how to Reproduce**
1. Pull the latest code on a Linux machine set up to run RAVEN
2. Run `./raven_framework interactive <input.xml>`.

**Screenshots and Input Files**
Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

No specific input is required. 

**Platform (please complete the following information):**
 - OS: Ubuntu, Fedora, Linux Mint
 - Version: 18.04(Ubuntu), 31(Fedora), 19.2(Linux Mint)
 - Dependencies Installation: Both Conda and PIP

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

**Comment by PaulTalbot-INL** [comment]
Hey, thanks for bringing this to our attention.

We've put a fix for this in PR #1098, so when that merges it should work again.

**Comment by alfoa** [comment]
Closing. It was supposed to be closed by #1098
